app-android/app/src/main/java/de/tutao/tutanota/AndroidMobileContactsFacade.kt

```diff
@@ -26,713 +43,724 @@ import kotlinx.serialization.json.Json
+					parseStoredContactData(entityCursor, storedContact)
+				}
+
+				if (entityCursor.getInt(7) == 1) {
+					storedContact.isDirty = true
+				}
+			}
+		}
+		return storedContact
+	}
+
+	private fun parseStoredContactData(entityCursor: Cursor, storedContact: AndroidContact) {
+		val mimeType = getCursorString(entityCursor, 3)
+		val data1 = getCursorString(entityCursor, 4)
+
+		when (mimeType) {
+			ContactsContract.CommonDataKinds.StructuredName.CONTENT_ITEM_TYPE -> {
+				storedContact.givenName = getCursorString(entityCursor, 5)
+				storedContact.lastName = getCursorString(entityCursor, 6)
+				storedContact.title = getCursorString(entityCursor, 8)
+				storedContact.middleName = getCursorString(entityCursor, 9)
+				storedContact.nameSuffix = getCursorString(entityCursor, 10)
+				storedContact.phoneticFirst = getCursorString(entityCursor, 11)
+				storedContact.phoneticMiddle = getCursorString(entityCursor, 12)
+				storedContact.phoneticLast = getCursorString(entityCursor, 13)
+			}
+
+			ContactsContract.CommonDataKinds.Email.CONTENT_ITEM_TYPE -> storedContact.emailAddresses.add(AndroidEmailAddress(data1, entityCursor.getInt(5), getCursorString(entityCursor, 6)))
+			ContactsContract.CommonDataKinds.Phone.CONTENT_ITEM_TYPE -> storedContact.phoneNumbers.add(AndroidPhoneNumber(data1, entityCursor.getInt(5), getCursorString(entityCursor, 6)))
+			ContactsContract.CommonDataKinds.StructuredPostal.CONTENT_ITEM_TYPE -> storedContact.addresses.add(AndroidAddress(data1, entityCursor.getInt(5), getCursorString(entityCursor, 6)))
+			ContactsContract.CommonDataKinds.Nickname.CONTENT_ITEM_TYPE -> storedContact.nickname = data1 ?: ""
+			ContactsContract.CommonDataKinds.Organization.CONTENT_ITEM_TYPE -> {
+				storedContact.company = data1
+				storedContact.role = getCursorString(entityCursor, 8)
+				storedContact.department = getCursorString(entityCursor, 9)
+			}
+
+			ContactsContract.CommonDataKinds.Event.CONTENT_ITEM_TYPE -> {
+				val type = entityCursor.getInt(5)
+				if (type == ContactsContract.CommonDataKinds.Event.TYPE_BIRTHDAY) {
+					storedContact.birthday = data1
+				} else {
+					storedContact.customDate.add(AndroidCustomDate(data1, type, getCursorString(entityCursor, 6)))
+				}
+			}
+
+			ContactsContract.CommonDataKinds.Relation.CONTENT_ITEM_TYPE -> storedContact.relationships.add(AndroidRelationship(data1, entityCursor.getInt(5), getCursorString(entityCursor, 6)))
+			ContactsContract.CommonDataKinds.Website.CONTENT_ITEM_TYPE -> storedContact.websites.add(AndroidWebsite(data1, entityCursor.getInt(5), getCursorString(entityCursor, 6)))
+			ContactsContract.CommonDataKinds.Note.CONTENT_ITEM_TYPE -> storedContact.notes = data1 ?: ""
 		}
-	  }
 	}
-	return storedContact
-  }
 
-  private fun parseStoredContactData(entityCursor: Cursor, storedContact: AndroidContact) {
-	val mimeType = entityCursor.getString(3)
-	val data1 = entityCursor.getString(4)
+	private fun getCursorString(cursor: Cursor, index: Int): String {
```

Comment:
I would probably make `fun Cursor.getNullableString()` or or `fun Cursor.getStringOrEmpty()` or something like that

Reply:
Thanks for the tip, I've implemented the function